lib/pbench/server/database/models/datasets.py

```diff
@@ -880,12 +896,27 @@ def validate(dataset: Optional[Dataset], key: str, value: Any) -> Any:
             maximum = base_time + datetime.timedelta(days=int(max_retention.value))
             if target > maximum:
                 raise MetadataBadValue(
-                    dataset, key, value, f"date/time before {maximum:%Y-%m-%d}"
+                    dataset, key, v, f"date/time before {maximum:%Y-%m-%d}"
                 )
             target += __class__.ONE_DAY
-            return f"{target:%Y-%m-%d}"
-        else:
-            return value
+            v = f"{target:%Y-%m-%d}"
+        elif key == __class__.ARCHIVEONLY:
+            if type(v) is str:
+                if v.lower() in ("t", "true", "y", "yes"):
+                    v = True
+                elif v.lower() in ("f", "false", "n", "no"):
+                    v = False
+                else:
+                    raise MetadataBadValue(dataset, key, v, "boolean")
+            elif type(v) is int:
+                v = bool(v)
```

Comment:
Sure; basically, 0 is false and anything else is true. I threw that in on a whim. Interestingly, that immediately caused me to break a test because unless you use `PUT /datasets/metadata/{id}` there's no way to specify a metadata key *value* as anything but "string", so writing `PUT /upload/{file}?metadata=server.archiveonly:0` doesn't actually work... so one might argue that including integer values is kinda pointless anyway. (Although we might later allow `PUT /datasets/metadata/{id} {"metadata": {"server.archiveonly": 1}}`...)

Reply:
> one might argue that including integer values is kinda pointless anyway

Since this parameter is supposed to be a boolean, if we cannot easily recognize the value as either true or false, then I recommend that we not support it.  To wit, I submit that `"0"` is ambiguous (it evaluates as `True` but is presumably intended to indicate `False`)...and, on that basis, I recommend that we not accept integer values at all.